mountpoint-s3-crt/src/auth/imds_client.rs

```diff
@@ -83,6 +83,29 @@ impl ImdsClient {
                 .ok_or_last_error()
         }
     }
+
+    /// Get the value for the given resource path.
+    pub fn get_resource<F>(&self, resource_path: &str, callback: F) -> Result<(), Error>
```

Comment:
Should we just use this for `get_instance_type` too? It seems semi-random to me which IMDS resources have methods in `aws_imds_client` and which don't, so maybe we should just have this one generic method instead?